### PR TITLE
Add histogram matching test

### LINKED ISSUE #4106: Histogram matching: test is missing for multichannel=False
## Description

Xref: https://github.com/scikit-image/scikit-image/issues/3618#issuecomment-448892397

### PR DESCRIPTION
**Pull request by robroooh**
## Description

This commit fixes #4106 by adding a testcase for match_histograms parameter.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/exposure/tests/test_histogram_matching.py [hunk 1]
```diff
@@ -27,8 +27,6 @@ class TestMatchHistogram:
     image_rgb = data.chelsea()
     template_rgb = data.astronaut()
 
-    # To handle with mutlichannel ==False
-    #(image_rgb[:, :, 0], template_rgb[:, :, 0]),
     @pytest.mark.parametrize('image, reference', [
         (image_rgb, template_rgb)
     ])
```
Comment (jni) [inline skimage/exposure/tests/test_histogram_matching.py]: I feel like this issue could be more easily solved with:

```python
@pytest.mark.parametrize('image, reference, multichannel', [
    (image_rgb, template_rgb, True),
    (image_rgb[..., 0], template_rgb[..., 0], False),
])
def test_match_histograms(image, reference, multichannel):
    ...
    matched = exposure.match_histograms(image, reference, multichannel=multichannel
```

Would that fix the problem or am I missing some other differences?

Comment (robroooh) [reply skimage/exposure/tests/test_histogram_matching.py]: Oh! Thank you for pointing it out. :smile: I was not familiar with parameterized testing. It's good to learn something new.

File: skimage/exposure/tests/test_histogram_matching.py [hunk 2]
```diff
@@ -27,17 +27,18 @@ class TestMatchHistogram:
     image_rgb = data.chelsea()
     template_rgb = data.astronaut()
 
-    # To handle with mutlichannel ==False
-    #(image_rgb[:, :, 0], template_rgb[:, :, 0]),
-    @pytest.mark.parametrize('image, reference', [
-        (image_rgb, template_rgb)
+    @pytest.mark.me
```
Comment (hmaarrfk) [inline skimage/exposure/tests/test_histogram_matching.py]:
```suggestion
```

**Comment by robroooh** [comment]
Thank you for pointing that out 👍. I didn't notice it in the first place.

**Comment by sciunto** [comment]
Thank you @robroooh  and @hmaarrfk

**Pull request merged**